### PR TITLE
Give `RemoteAsset` a `created` attribute

### PR DESCRIPTION
**Pull request by jwodder**
Motivated by https://github.com/dandi/dandisets/issues/84.

### CODE REVIEW

File: dandi/dandiapi.py [hunk 1]
```diff
@@ -1116,6 +1116,9 @@ class RemoteAsset(BaseRemoteAsset):
     #: The identifier for the version of the Dandiset to which the asset
     #: belongs
     version_id: str
+    #: The date at which the asset was created.  This is not set if the asset
+    #: was returned by `RemoteDandiset.get_asset()`.
```
Comment (yarikoptic) [inline dandi/dandiapi.py:1120]: when is it set?

Comment (yarikoptic) [reply dandi/dandiapi.py:1120]: should it become some lazy property so it could be queried if not known but requetsed?

Comment (jwodder) [reply dandi/dandiapi.py:1120]: @yarikoptic It's set when it's returned by any of the other methods that return `RemoteAsset`s.  The only way to query it if not known is by iterating through all the assets.

Comment (yarikoptic) [reply dandi/dandiapi.py:1120]: There is path argument to dandisets_versions_assets_list (didn't try) - do may be could be limited to just that asset?

Comment (jwodder) [reply dandi/dandiapi.py]: @yarikoptic The `get_asset()` method now makes a second call to dandiset_versions_assets_list in order to get the value for the `created` field.  It also uses this call to populate the `modified` field, which has been removed from the `BaseRemoteAsset` class, as there it was derived from the metadata, which need not agree with the asset listing value.

**Pull request merged**